app/src/hooks/useKeycloakWrapper.tsx

```diff
@@ -136,19 +147,45 @@ function useKeycloakWrapper(): IKeycloakWrapper {
   }
 
   /**
-   * Parses out the username portion of the preferred_username from the token.
+   * Coerces a string into a user identity source, e.g. BCEID, IDIR, etc.
+   *
+   * @example _inferIdentitySource('idir') => SYSTEM_IDENTITY_SOURCE.IDIR
+   *
+   * @param userIdentitySource The user identity source string
+   * @returns {*} {SYSTEM_IDENTITY_SOURCE | null}
+   */
+  const _inferIdentitySource = (userIdentitySource: string | undefined): SYSTEM_IDENTITY_SOURCE | null => {
+    switch (userIdentitySource) {
+      case SYSTEM_IDENTITY_SOURCE.BCEID_BASIC:
+        return SYSTEM_IDENTITY_SOURCE.BCEID_BASIC;
+
+      case SYSTEM_IDENTITY_SOURCE.BCEID_BUSINESS:
+        return SYSTEM_IDENTITY_SOURCE.BCEID_BUSINESS;
+
+      case SYSTEM_IDENTITY_SOURCE.IDIR:
+        return SYSTEM_IDENTITY_SOURCE.IDIR;
+
+      default:
+        return null;
+    }
+  };
+
+  /**
+   * Parses out the username from a keycloak token, from either the `idir_username` or `bceid_username` field.
    *
    * @param {object} keycloakToken
    * @return {*} {(string | null)}
    */
   const getUserIdentifier = useCallback((): string | null => {
-    const userIdentifier = keycloakUserDataLoader.data?.['preferred_username']?.split('@')?.[0];
+    const userIdentifier =
+      (keycloakUserDataLoader.data as IIDIRUserInfo)?.idir_username ||
+      (keycloakUserDataLoader.data as IBCEIDBasicUserInfo)?.bceid_username;
```

Comment:
Is this type cast to bceid-basic safe if its a bceid-business user?

Reply:
yes because the BCEID business interface extends the Basic one.